sentry_streams/src/gcs_writer.rs

```diff
@@ -55,35 +59,47 @@ impl GCSWriter {
 
         let client = client.default_headers(headers).build().unwrap();
 
-        GCSWriter { client, url }
+        GCSWriter { client, url, route }
     }
+}
 
-    pub fn write_to_gcs(
-        &self,
-        message: Message<RoutedValue>,
-    ) -> Result<Message<RoutedValue>, SubmitError<RoutedValue>> {
+impl TaskRunner<RoutedValue, RoutedValue, anyhow::Error> for GCSWriter {
+    fn get_task(&self, message: Message<RoutedValue>) -> RunTaskFunc<RoutedValue, anyhow::Error> {
         let client = self.client.clone();
         let url = self.url.clone();
-        let bytes = Python::with_gil(|py| pybytes_to_bytes(&message, py)).unwrap();
+        let route = message.payload().route.clone();
+        let actual_route = self.route.clone();
+
+        let bytes = traced_with_gil("writing to gcs", |py| pybytes_to_bytes(&message, py)).unwrap();
 
-        let res = client.post(&url).body(bytes).send();
+        Box::pin(async move {
+            // TODO: This route-based forwarding does not need to be
+            // run with multiple threads. Look into removing this from the async task.
+            if route != actual_route {
+                return Ok(message);
+            }
```

Comment:
Actually it would be trivial to fix it here. 
As you reintroduced the outer GCSSink strategy you can put this logic in the `submit` method there avoiding to pass it to inner.

Reply:
Or is the issue due to the fact you cannot have two owners of the next step?

Reply:
This

Reply:
@untitaker Did you all ever run into this in Rust Arroyo? Do yo know of any workarounds? 

Basically I want to be able to send messages on a different route to the next step (after `RunTaskInThreads`). The issue is that `RunTaskInThreads` step owns next_step

Reply:
you can add an accessor on `RunTaskInThreads` such as:

```
fn next_step_mut(&mut self) -> &mut N { &mut self.next_step }
```

then you can use `self.inner.next_step_mut().submit(...)` in your GCS Sink

I don't think this usecase of poking into the next step has come up before, but i did find it occasionally annoying for observability that one step completely hides the next one. might be worth adding to the strategy interface.